src/averageAges.js

```diff
@@ -55,7 +76,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const relevantChildren = (
+    people.filter(person => onlyWithSon
+      ? people.some(child => person.mother === child.name) && person.sex === 'm'
```

Comment:
```suggestion
      ? people.some(parent => person.mother === parent.name) && person.sex === 'm'
```

as I understand this is not a `child`, but `parent` 🤔